### ISSUE TITLE
Jazz Pakistan VoWifi not working

### ISSUE THREAD

**Issue by sirfnomi** [kyujin-cho/pixel-volte-patch#171]
Hi, 

first of all thank you so much for this patch which helps me to enable VoLTE service but unfortunately VoWifi service is showing as unavailable.
my network is Jazz Pakistan.
I already tried with the new APN name as ims but the results are the same.
is there any further settings or modification we can do for VoWifi service ?
I have another device which is Samsung which has VoLTE and VoWifi working if any setting on this device is useful I can provide. 

thanks again for your work

**Comment by Sanshe16** [comment]
Call jazz helpline to activate VoWifi and Volte on your number. I am using both on OnePlus 9 pro.

**Comment by sirfnomi** [comment]
Hi Sanshe,

VoWiFi and VoLTE are already activated on my number as I verified it by inserting my SIM in another phone in which both services are working officially from Jazz.

Can you share your apn screenshot with me ?

**Comment by Sanshe16** [comment]
![IMG_20230724_051711.jpg](https://github.com/kyujin-cho/pixel-volte-patch/assets/54706835/ee9f599b-ac8e-46a2-ab26-aa730049c6cd)

![IMG_20230724_051733.jpg](https://github.com/kyujin-cho/pixel-volte-patch/assets/54706835/5ab001af-5f8d-4d34-a971-8d9b8aa41da0)

**Comment by sirfnomi** [comment]
I will ask them again and see if they agree to help me with this issue.

Thank you for screenshot but please share Access Point Name screen with details page.
Actually I'm looking for access point name with ims name 

![255439165-5ab001af-5f8d-4d34-a971-8d9b8aa41da0~2](https://github.com/kyujin-cho/pixel-volte-patch/assets/11202594/3ee07183-0abc-41f5-afbc-a84e16b86ae1)

**Comment by Sanshe16** [comment]
Also check this link as well.

https://www.pakgamers.com/index.php?threads/the-official-3g-4g-lte-thread.232438/post-3033393

![Screenshot_2023-07-24-12-32-11-82_426a5588c5110cd01d5af17e47adb223.jpg](https://github.com/kyujin-cho/pixel-volte-patch/assets/54706835/386eb2b8-a418-46bc-bd5f-a41fecfb3101)

![Screenshot_2023-07-24-12-32-07-87_426a5588c5110cd01d5af17e47adb223.jpg](https://github.com/kyujin-cho/pixel-volte-patch/assets/54706835/ce024aa5-03dd-40eb-914d-c81ef0497aa4)

**Comment by sirfnomi** [comment]
Thank you for the Screenshots Sanshe, I'm grateful to you.
I'm following that forum.

**Comment by xinseo** [comment]
> Thank you for the Screenshots Sanshe, I'm grateful to you. I'm following that forum.

Any luck? facing the same issue

**Comment by sirfnomi** [comment]
your mobile ?

**Comment by xinseo** [comment]
> your mobile ?

Pixel 7 non-rooted

**Comment by xinseo** [comment]
> your mobile ?

any luck?

**Comment by sirfnomi** [comment]
> > your mobile ?
> 
> Pixel 7 non-rooted on Jazz.

Share your ims registration status.

I confirm that Jazz VoWifi is working on Pixel 7 using this patch.
My cousin is using VoWifi on Pixel 7, you need to ask the helpline for service activation on your number or for auto activation you need to temporarily insert your SIM in a supported mobile phone and after enabling VoWifi service then you can insert your sim back in Pixel 7.
Make sure VoWifi enabled in sim setting menu.

**Comment by xinseo** [comment]
> > > your mobile ?
> > 
> > 
> > Pixel 7 non-rooted on Jazz.
> 
> Share your ims registration status.
> 
> I confirm that Jazz VoWifi is working on Pixel 7 using this patch. My cousin is using VoWifi on Pixel 7, you need to ask the helpline for service activation on your number or for auto activation you need to temporarily insert your SIM in a supported mobile phone and after enabling VoWifi service then you can insert your sim back in Pixel 7. Make sure VoWifi enabled in sim setting menu.

I actually have a jazz eSIM.

**Comment by sirfnomi** [comment]
Did you install this patch?

Dial the code on phone app *#*#4636#*#* New window will open then tap on phone info then tap on IMS registration status from menu and share this screenshot.

**Comment by xinseo** [comment]
> Did you install this patch?
> 
> Dial the code on phone app _#_#4636#_#_ New window will open then tap on phone info then tap on IMS registration status from menu and share this screenshot.

![Screenshot_20230808-093248](https://github.com/kyujin-cho/pixel-volte-patch/assets/6553354/2ae8c9d3-6c9d-41fe-9510-f62b8717b8ea)

**Comment by sirfnomi** [comment]
Call the helpline and ask to activate these services. Like VoLTE and VoWiFi

**Comment by xinseo** [comment]
> Call the helpline and ask to activate these services. Like VoLTE and VoWiFi

They have enabled both services. This is what the screenshot shows now.

![Screenshot_20230808-145946](https://github.com/kyujin-cho/pixel-volte-patch/assets/6553354/ee756c68-c44a-4e78-8363-646509458991)


VoWiFi is still shown as Unavailable. I am not connected to a Wi-Fi network right now. Do I need to be connected to a Wi-Fi network for it to show as Available?

**Comment by sirfnomi** [comment]
Yes you need wifi

**Comment by xinseo** [comment]
> Yes you need wifi

Thank you for the the help. Working perfectly now.

**Comment by sirfnomi** [comment]
Glad 😊

**Comment by kodi1122** [comment]
VoLTE and VoWIFI both working on Pixel 6 thanks. 
![Uploading Screenshot_20241214-183644.png…]()

**Comment by botisgoat** [comment]
> Glad 😊

Hey, I am using pixel 8 pro with Jazz Physical SIM. I just enabled VoWiFi using IMS tool. But when I enable WiFi calling from the SIM settings the IMS registration disappears and when I disable WiFi Calling it comes back. I have confirmed by calling jazz helpline that VoWiFi and VoLTE is enabled on my SIM. Can you help me how can I fix this please?

**Comment by sirfnomi** [comment]
Can you share more information, such as your IMS status as depicted in the screenshot above? Have you added the APN setting for IMS?

**Comment by botisgoat** [comment]
> Can you share more information, such as your IMS status as depicted in the screenshot above? Have you added the APN setting for IMS?


I just modified the APN settings. I didn't see IPv4 at first, but now that I've enabled it, everything is working perfectly. Thanks for your help! I can now use VoWiFi on the Jazz network with my Pixel 8 Pro.

**Comment by sirfnomi** [comment]
Enjoy your VoWiFi calls!

**Comment by 0xsudoSaint** [comment]
@sirfnomi , whats up with the "server unreachable" problem. Have you  guys faced such issue?

**Comment by sirfnomi** [comment]
I haven't run into this issue myself.
The "server unreachable" means your phone is having trouble connecting to Jazz's servers. It could be a problem with a weak or unstable Wi-Fi connection.

**Comment by 0xsudoSaint** [comment]
Cant be with phone, neither with wifi. I have a real good internet connection (and its stable) , and phone is supported as well. I don't know, after my comment ,it was working fine during night.

**Comment by sirfnomi** [comment]
alright !
do you use VPN ?

**Comment by 0xsudoSaint** [comment]
Hmm, I guess it was VPN after all. But, on the other device. I don't know, the traffic from that device should not affect VoWifi transmission from mobile. Well, thanks to you i diagnosed it.

**Comment by sirfnomi** [comment]
Glad 👍

**Comment by Abdu11ahBilal** [comment]
@sirfnomi i'm using a pixel 8 
Carrier is jazz in Pakistan 
I've enabled volte from the settings as well as the pixel ims app but the ims status is still unregistered and calls automatically fall back to 2G 
I tried making a separate apn named ims but it said duplicate which means it already exists. 
Any ideas on what's the issue here ?

**Comment by botisgoat** [comment]
@Abdu11ahBilal this patch stopped working for me so I stopped using it and found a new patch which works.
https://github.com/vvb2060/Ims
You just have to allow it through shizuku and start the shizuku service once and the patch will be applied. You will have to apply it again by starting shizuku after every restart or update cause it is removed after it.

**Comment by Abdu11ahBilal** [comment]
@S3xy-B0t I've tried it but as soon as I allow ims via shizuku the app itself simply disappears from the authorized apps menu 
I restarted my phone and it was there again but still no VoLTE 
I even tried another app called turbo ims 
But it crashes as soon as I hit apply settings to sim 
I dunno how it's working for everyone else but me

**Comment by sirfnomi** [comment]
VoLTE is officially supported now on Pixel phones for Pakistani networks like Jazz and Zong. There is no need to use this or any patch for VoLTE service.
If VoLTE is not working on your mobile, ensure that it is enabled on your SIM by your network provider.
and make sure your mobile is updated to latest software.

**Comment by Abdu11ahBilal** [comment]
I've got a pixel 8 and it's not working on it for some reason. I even bought a new sim today just to see if it may fix the issue. 
I called my carrier customer support 
They said it's enabled for my sim and works fine from their end. 
I'm on the latest January 2026 version of Android 16

**Comment by sirfnomi** [comment]
If Jazz says VoLTE is enabled on your number, the issue is likely with your device. To pinpoint the problem,  check the following.

* Try your SIM in another VoLTE-compatible phone to see if the service works there.
* Check your software version. Google officially released an update about 6 months ago that includes VoLTE support.
* Dial *#*#4636#*#*, go to 'Phone Information', and check if the IMS status shows as 'Registered'.

I’m using VoLTE on Jazz without any patches, and my friends with Pixel phones are too, so it should work for you. If it still doesn't work, patching is the only option. However, as @S3xy-B0t mentioned, on the latest Google updates, this IMS patch only lasts until the phone reboots. You’ll have to re-apply it every time you restart.

But first try if it works officially.

**Comment by Abdu11ahBilal** [comment]
@sirfnomi @S3xy-B0t 
I've literally tried every method and I dunno what I'm doing wrong so here's the stuff I've tried. Tell me what I've missed 

​Location: Lahore, Pakistan
Device: Google Pixel 8
OS Build: Android 16 (Build: Jan 05, 2026)
Carriers Tested: Jazz (Prepaid/Postpaid), Zong
Status: IMS Unregistered / VoLTE Toggle Visible but Non-Functional

​Troubleshooting History (All Methods Attempted)

​1. Software & Patching Overrides

​Shizuku + Pixel IMS/Turbo IMS: 
Attempted to force VoLTE/VoWiFi flags.  

​Result:
Toggles appeared in settings, but IMS Service Status remained "Not Registered."

​Pixel IMS v3.0 (No-GUI Method): 
Used the latest "Instrumentation" based patcher to bypass the late-2025 security blocks.

​Result: 
App successfully authorized in Shizuku but failed to register the IMS handshake.

​Activity Launcher (Hidden Activities): 
Attempted to access com.google.android.carrierconfig and Modem Test menus.

​Result: Most activities are now restricted or greyed out in the Jan 2026 build.

​2. Network & Carrier Provisioning

​APN Manual Configuration: 
* Modified Jazz APN to include ims,ia.
​Attempted to create a standalone "IMS" APN (System returned "Duplicate APN" error, indicating a hidden system profile).
​Switched APN Protocol from IPv4/IPv6 to IPv4 Only.

​Carrier SMS Activation:

​Texted VoLTE to 7342 (Jazz).
​Response: "Your phone does not support this OTA." (Indicates server-side IMEI rejection).

​Texted REG to 431 (Zong).
​Result: Handshake failed to transition to "Registered" status.

​SIM Hardware Swaps:
​Purchased a brand new Jazz SIM to rule out legacy SIM card corruption.
​Tested friend's Zong SIM (previously active on VoLTE).
​Result: Both failed to register IMS on the Pixel 8.

​3. Modem & Radio Tweaks (*#*#4636#*#*)

​Preferred Network Type: Cycled through LTE Only, NR/LTE, and GSM/WCDMA/LTE.
​Result: "LTE Only" killed all calling functionality, confirming no IMS fallback path.
​Radio Cycle: Performed "Turn Off Radio" -> "Turn On Radio" sequence.
​Result: Temporary signal refresh but IMS status remained "Not Registered."

​IMS Hidden Menu: Attempted to modify NR_TIMER and IMS_REGISTRATION_TIMER via *#*#0702#*#*.
​Result: No change in registration behavior.

​Technical Blocker: The "Persistence Loop"
​On the January 05, 2026 Android 16 build, we are observing a critical failure in the patch-cycle:

​The Reboot Trap: 
To apply modem-level patches, a reboot is traditionally required.

​The Security Seal: 
Upon reboot, the Shizuku "Shell" is killed. Because the Jan 2026 patch has hardened MODIFY_PHONE_STATE permissions, the "laundering" script cannot maintain persistence long enough to bridge the gap between "Boot" and "Network Attach."

​Greyed-Out Provisioning: 
In the testing menu, "VoLTE Provisioned" is toggled ON but remains GREYED OUT, signifying that the OS has the flag set, but the Modem is being blocked by the Carrier's MBN (Modem Binary) profile.

**Comment by botisgoat** [comment]
@Abdu11ahBilal My friend has a Pixel 7 pro with tampered IMEI and it is a Japanese variant. He also has the same issue. VoLte and VoWifi does not work on his phone with or without patch. His IMS status also remains unregistered. I have 8 pro with original IMEI and US variant and patch works for me. Do you have any similarity with my friend's case.

**Comment by Abdu11ahBilal** [comment]
@S3xy-B0t 
My pixel 8 has original IMEI and it's pta approved but the only similar thing is that it's also a japanese variant like ur cousin's

**Comment by botisgoat** [comment]
@Abdu11ahBilal I am not sure but I think the issue lies in the phone being Japanese variant.

**Comment by Abdu11ahBilal** [comment]
@S3xy-B0t Yup i second that conclusion since my older 6a was also a Japanese variant and had the same issues. NOTHING worked on it as well. Add ur cousin's 7 pro and it cements the fact that Japanese variants specifically hv this issue. I looked up on reddit and a guy in Sri Lanka with a Jap variant seems to hv the same issue. I think i should make a separate issue for this.

**Comment by salman0x00** [comment]
IOS wifi calling Working Telenor PK

<img width="603" height="1311" alt="Image" src="https://github.com/user-attachments/assets/57a3a9b3-f7fd-480f-af8e-a0f3eeae06ca" />

**Comment by sirfnomi** [comment]
Great! Is Telenor the first Pakistani network to officially support VoWiFi on iOS, or is Apple provisioning VoWiFi for all Pakistani networks? I noticed in your screenshot that Jazz isn't showing VoWiFi.

**Comment by matifali** [comment]
I also own a Pixel 7 Pro using Jazz. I loose IMS registration after each reboot

**Comment by hassanwaqas** [comment]
Hi Everyone. Just going through the thread for my own same issue. I was able to finally get it done.
If you have already applied IMS patch with shizuku and still vowifi isnt working, then just manually create an APN under Jazz Sim settings. The name of APN should be IMS as its created in official Jazzfi enabled phones in pakistan. APN Name: IMS, APN type : IMS , Protocols etc : IPv4/IPv6 . Just replicate the settings from some samsung model and vowifi would start working. i did it in this way. Hope it helps you guys

**Comment by matifali** [comment]
@hassanwaqas, for me, turning on VoWiFi in Pixel IMS crashes it and disables my VoLTE too. I already have the APN setting for IMS that made VoLTE work initially.

**Comment by sirfnomi** [comment]
@matifali use the latest or beta build of IMSPixel and toggle off the persistent change. (you have to repeat these steps every time when you restart the phone.)

**Comment by matifali** [comment]
Already using the latest built. Also have added the ims APN. VoLTE works but if I enable VoWiFi VoLTE breaks too.

**Comment by botisgoat** [comment]
@matifali Are you using this patch? https://github.com/vvb2060/Ims/releases/tag/v3.0

**Comment by hassanwaqas** [comment]
> [@hassanwaqas](https://github.com/hassanwaqas), for me, turning on VoWiFi in Pixel IMS crashes it and disables my VoLTE too. I already have the APN setting for IMS that made VoLTE work initially.

Please use following patch which remains persistent even after restarting. Just disable the volte pixel patch of this thread from shizuku and grant permission to below one from shizuku. It works fine with IMS APN along with main internet APN (Currently the pixel phones fetch main APN with GPRS). Instead go for both the two APNs as present in samsung phones.

https://github.com/vvb2060/Ims/releases/tag/v3.0

**Comment by matifali** [comment]
![Screenshot_20260331-172809.png](https://github.com/user-attachments/assets/12706e6c-631a-46b0-88c0-6523192b444a)

![Screenshot_20260331-172804.png](https://github.com/user-attachments/assets/0cef8f9a-da9f-4e0e-a7db-a8d70fc18662)

See the screenshots. As soon as I toggle VoWifi On I loose VoLTE too and IMS Registration status becomes Unregistered. 

> @matifali Are you using this patch? https://github.com/vvb2060/Ims/releases/tag/v3.0

Yes @botisgoat I am using this patch.

**Comment by RamzanZahoor** [comment]
Hey, anyone's Vowifi working or not? Mine not working from 3 months. And it's a hectic.

**Comment by botisgoat** [comment]
@RamzanZahoor Yup mine is working perfectly.

**Comment by botisgoat** [comment]
@RamzanZahoor https://github.com/kyujin-cho/pixel-volte-patch
This one

**Comment by RamzanZahoor** [comment]
Man....this is amazing. At least found someone who have it working.

1. Can you please tell me which Pixel IMS app are you using?
2. Which model(Pixel 6, 7, or 8) you have?
3. How you added your APN settings to get "registered IMS status"?
4. What are the settings you enabled/disabled in your Pixel IMS app.


Really feeling very desperate to know about it.

Thanks for your attention.

**Comment by botisgoat** [comment]
@RamzanZahoor 
1. I am using pixel ims by kyujin-cho. https://github.com/kyujin-cho/pixel-volte-patch
2. I am using pixel 8 pro (US variant).
3. I am using Zong (e-sim) and Jazz physical sim. I have added APN with name, APN, and APN type fields as "ims". APN protocol and APN roaming protocol as IPv4 and network type is unspecified rest of the fields are empty. Zong sim is working without adding the APN.
4. I have enabled VoLTE, VoNR, VoWiFi, Video calling (VT), Enable Enhanced 4g LTE (LTE+) and Allow adding APNs.

**Comment by RamzanZahoor** [comment]
Thank you so much.

I have added everything. But it's still showing IMS status "unregistered"
for my jazz sim.

Let me share it to you:

This is the APN - You mentioned to keep the APN name as "APN" , I kept it,
and it didn't work. I also then changed it to Name: "IMS" and APN: "IMS"



Then on Shizuku side it's still:


No matter what I do, it's remaining Unregistered.

I have reset my network settings for Jazz as well, and repeated all the
process you mentioned but it's still not working.

Thanks you so much for your time and...

**Comment by botisgoat** [comment]
@RamzanZahoor These are my APN settings.
![Screenshot_20260615-004024.png](https://github.com/user-attachments/assets/a2e79636-971c-428c-a606-d960cdf9e7cc)

**Comment by RamzanZahoor** [comment]
Nothing changed unfortunately even after changing Name : "ims" and APN:
"ims" as well.



It's still IMS status unregistered.
But thanks for your time🙂

**Comment by sirfnomi** [comment]
Hi @RamzanZahoor ,

Have you confirmed with your network provider that VoLTE and VoWiFi services are actively provisioned on your specific number? VoLTE service is now officially supported for Pixel phones across Pakistani carriers, so there is no need for any patch for VoLTE service, they often require manual activation by the provider. If it is activated on your number, IMS registration should show as registered.

To help us troubleshoot this further, could you please provide the following details:

1. Which Pixel model and variant are you using?
2. which android version and build number are you on ?
3. Have you tested your SIM in another device, or tested a different SIM in your Pixel? 
4. Please share a screenshot of your IMS registration status?.
5. Could you please share a screenshot of your current APN settings for the Jazz network?

**Comment by imfaxli** [comment]
Pixel 9 pro cpid 
Android 17
Jazz sim
Volte and vowifi not working enhanced lte working 
No ims

<img width="960" height="358" alt="Image" src="https://github.com/user-attachments/assets/194e0ce9-2340-48ef-8695-081354459f95" />
<img width="960" height="4207" alt="Image" src="https://github.com/user-attachments/assets/37bb2306-794a-46fe-96ce-571f3d6afaa3" />
<img width="960" height="2142" alt="Image" src="https://github.com/user-attachments/assets/1391af96-03ee-4c66-9ac5-0515f819c5c8" />

**Comment by imfaxli** [comment]
Thanks it was Jazz end issue
I've called customer care and told them disable and re enable the volte service and it worked
